File: perl5/optimize-media.pl
#!/usr/bin/perl
# optimize-media.mjs — portiert nach perl5
# Quelle: javascript, Onboarding@main:scripts/optimize-media.mjs
# Erzeugt: 2026-08-19 durch ABSTRACTIONS_MANAGER.py

use strict;
use warnings;
use File::Find;
use File::Basename;
use Image::Magick;

# Das Verzeichnis ../public/media/ relativ zum Skript bestimmen
my $script_dir = dirname(__FILE__);
my $media_dir = "$script_dir/../public/media/";

# Überprüfen, ob das Verzeichnis existiert
unless (-d $media_dir) {
    die "Verzeichnis $media_dir nicht gefunden.\n";
}

# Alle PNG-Dateien im Verzeichnis finden
opendir(my $dh, $media_dir) or die "Konnte Verzeichnis $media_dir nicht öffnen: $!";
my @files = grep { /\.png$/i } readdir($dh);
closedir $dh;

foreach my $file (@files) {
    my $source_path = "$media_dir/$file";
    my ($stem) = $file =~ /^(.+)\.png$/i;

    # WebP erstellen
    my $webp_path = "$media_dir/${stem}.webp";
    convert_image($source_path, $webp_path, 'WEBP', 84);

    # AVIF erstellen
    my $avif_path = "$media_dir/${stem}.avif";
    convert_image($source_path, $avif_path, 'AVIF', 58);
}

print "WebP- und AVIF-Derivate erzeugt.\n";

sub convert_image {
    my ($source, $dest, $format, $quality) = @_;

    my $image = Image::Magick->new();
    my $x = $image->Read($source);
    warn "Fehler beim Lesen von $source: $x" if $x;

    $x = $image->Set(compression => 'Lossless') if $format eq 'AVIF';
    $x = $image->Write(filename => $dest, quality => $quality);
    warn "Fehler beim Schreiben von $dest: $x" if $x;
}

File: python/optimize-media.py
#!/usr/bin/env python3
# optimize-media.mjs — portiert nach python
# Quelle: javascript, Onboarding@main:scripts/optimize-media.mjs
# Erzeugt: 2026-08-19 durch ABSTRACTIONS_MANAGER.py

import asyncio
import os
from pathlib import Path
from urllib.parse import urljoin
from urllib.request import url2pathname
from urllib.parse import urlparse
import aiofiles
from PIL import Image

async def main():
    # Bestimme das Verzeichnis relativ zum Skriptverzeichnis
    script_dir = Path(__file__).parent
    media_dir = script_dir.parent / 'public' / 'media'
    
    # Liste alle Dateien im Verzeichnis auf
    files = os.listdir(media_dir)
    
    for file in files:
        if not file.endswith(".png"):
            continue
            
        source_path = media_dir / file
        stem = Path(file).stem
        
        # Konvertiere zu WebP
        webp_path = media_dir / f"{stem}.webp"
        with Image.open(source_path) as img:
            img.save(webp_path, 'webp', quality=84)
        
        # Konvertiere zu AVIF (falls unterstützt)
        avif_path = media_dir / f"{stem}.avif"
        with Image.open(source_path) as img:
            img.save(avif_path, 'avif', quality=58)
    
    print("WebP- und AVIF-Derivate erzeugt.")

if __name__ == "__main__":
    asyncio.run(main())
